src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceCounterLib.cs

```diff
@@ -1182,8 +1203,21 @@ internal static void RegisterCategory(string categoryName, PerformanceCounterCat
                     CreateSymbolFile(creationData);
                     RegisterFiles(IniFilePath, false);
                 }
-                CloseAllTables();
-                CloseAllLibraries();
+                /*
+                  Wait some seconds for publication, sometimes registry modification is not immediatly visibile to all.
+                  (https://msdn.microsoft.com/en-us/library/cs38wsc4%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396)
+                  Similar pattern of GetStringTable()  "int waitRetries = 14;   //((2^13)-1)*10ms == approximately 1.4mins"
+                 */
+                bool isPublished = false;
+                DateTime now = DateTime.UtcNow;
+                while ((DateTime.UtcNow.Subtract(now).TotalSeconds < 10))
+                {
+                    CloseAllLibraries();
```

Comment:
It looks like CloseAllTables got removed.  Is that intentional?  I'd expect that you'd want completely clean state.

Reply:
CloseAllLibraries() call library.Close() that call CloseTables()

Reply:
Ok, thank you.